test/integration/files/FilesRepository.test.ts

```diff
@@ -0,0 +1,227 @@
+import { createDatasetViaApi } from '../../testHelpers/datasets/datasetHelper';
+import { uploadFileViaApi } from '../../testHelpers/files/filesHelper';
+import { FileOrderCriteria } from '../../../src/files/domain/models/FileOrderCriteria';
+import { DatasetsRepository } from '../../../src/datasets/infra/repositories/DatasetsRepository';
+import { ReadError } from '../../../src/core/domain/repositories/ReadError';
+
+describe('FilesRepository', () => {
+  const sut: FilesRepository = new FilesRepository();
+
+  const testTextFile1Name = 'test-file-1.txt';
+  const testTextFile2Name = 'test-file-2.txt';
+  const testTextFile3Name = 'test-file-3.txt';
+  const testTabFile4Name = 'test-file-4.tab';
+
+  const nonExistentFiledId = 200;
+
+  beforeAll(async () => {
+    ApiConfig.init(TestConstants.TEST_API_URL, DataverseApiAuthMechanism.API_KEY, process.env.TEST_API_KEY);
+    await createDatasetViaApi()
+      .then()
+      .catch(() => {
+        fail('Test beforeAll(): Error while creating test Dataset');
+      });
+    // Uploading test file 1
+    await uploadFileViaApi(TestConstants.TEST_CREATED_DATASET_ID, testTextFile1Name)
+      .then()
+      .catch((e) => {
+        console.log(e);
+        fail(`Tests beforeAll(): Error while uploading file ${testTextFile1Name}`);
+      });
+    // Uploading test file 2
+    await uploadFileViaApi(TestConstants.TEST_CREATED_DATASET_ID, testTextFile2Name)
+      .then()
+      .catch((e) => {
+        console.log(e);
+        fail(`Tests beforeAll(): Error while uploading file ${testTextFile2Name}`);
+      });
+    // Uploading test file 3
+    await uploadFileViaApi(TestConstants.TEST_CREATED_DATASET_ID, testTextFile3Name)
+      .then()
+      .catch((e) => {
+        console.log(e);
+        fail(`Tests beforeAll(): Error while uploading file ${testTextFile3Name}`);
+      });
+    // Uploading test file 4
+    await uploadFileViaApi(TestConstants.TEST_CREATED_DATASET_ID, testTabFile4Name)
+      .then()
+      .catch((e) => {
+        console.log(e);
+        fail(`Tests beforeAll(): Error while uploading file ${testTabFile4Name}`);
+      });
+  });
+
+  describe('getDatasetFiles', () => {
+    describe('by numeric id', () => {
+      test('should return all files filtering by dataset id', async () => {
+        const actual = await sut.getDatasetFiles(TestConstants.TEST_CREATED_DATASET_ID);
+        assert.match(actual.length, 4);
+        assert.match(actual[0].name, testTextFile1Name);
+        assert.match(actual[1].name, testTextFile2Name);
+        assert.match(actual[2].name, testTextFile3Name);
+        assert.match(actual[3].name, testTabFile4Name);
+      });
+
+      test('should return correct files filtering by dataset id and paginating', async () => {
+        const actual = await sut.getDatasetFiles(TestConstants.TEST_CREATED_DATASET_ID, undefined, 3, 3, undefined);
+        assert.match(actual.length, 1);
+        assert.match(actual[0].name, testTabFile4Name);
+      });
```

Comment:
In a later PR I include the FileCriteria object that groups all the different criteria for getting files.

I decided not to test all possible filtering scenarios, combining the optional parameters in different ways as you propose, since all those tests are already implemented on the Dataverse API side. If you are interested, see: https://github.com/IQSS/dataverse/blob/35e454767288d191d496b04a2ee0f6471bebac53/src/test/java/edu/harvard/iq/dataverse/api/DatasetsIT.java#L3300

For js-dataverse, I find enough to check that the parameters sent by the module are correctly interpreted by the API, whatever the combination of them.

Reply:
> I decided not to test all possible filtering scenarios, combining the optional parameters in different ways as you propose, since all those tests are already implemented on the Dataverse API side. If you are interested, see: https://github.com/IQSS/dataverse/blob/35e454767288d191d496b04a2ee0f6471bebac53/src/test/java/edu/harvard/iq/dataverse/api/DatasetsIT.java#L3300

Sorry, I was thinking this was a unit test. 

Anyway, in the unit tests we don't have these tests either. If we offer the user the possibility to pass undefined to one of those arguments then I think we should check those cases work, and I'm not thinking on testing the API, I'm just thinking on testing the request creation that could be malformed when passing an undefined to one of the arguments of the use case